### ISSUE TITLE
Use `fetch` from load event

### ISSUE THREAD

**Issue by david-plugge** [icflorescu/trpc-sveltekit#54]
**Describe the bug**
The fetch function provided by the event inside load function deduplicates api calls by inlining the reponse.
The first fetch on the client will get that inlined data.

**Expected behavior**
No extra api call on the first client side load.

**Additional context**
This will result in calling `createTRPCClient` in every load function, not sure it that´s a big deal.

**Comment by david-plugge** [comment]
After a bit of testing i found out that passing the origin in the url disables the inlining.

I modified the `createTRPCClient` function a bit and got it working:

```ts
function createTRPCClient<Router extends AnyRouter>(...) {
	let link: TRPCLink<Router>;

	if (init) {
		const {
			fetch,
			url: { origin }
		} = init;

		link = httpBatchLink({ url: `${url}`, fetch, headers });
	} else {
		if (typeof window === 'undefined') {
			throw new Error(
				'Calling createTRPCClient() on the server requires passing a valid LoadEvent argument'
			);
		}
		link = httpBatchLink({ url: `${url}`, headers });
	}

	// eslint-disable-next-line @typescript-eslint/ban-ts-comment
	// @ts-ignore
	return createTRPCProxyClient<Router>({ transformer, links: [link] });
}
```

**Comment by icflorescu** [comment]
Thanks for the tip! This should be fixed in `v3.2.5`.
